src/audio/sequence.c

```diff
@@ -1,31 +1,922 @@
+            return false;
+        }
+    }
+
+    return true;
+}
+
+// Unused
+void AudioSeq_ResetSequenceRequests(u8 seqPlayerIndex) {
+    sNumSeqRequests[seqPlayerIndex] = 0;
+}
+
+/**
+ * Check if the setup command is queued. If it is, then replace the command
+ * with `SEQCMD_SUB_OP_SETUP_RESTORE_SEQPLAYER_VOLUME`.
+ * Unused
+ */
+void AudioSeq_ReplaceSeqCmdSetupOpVolRestore(u8 seqPlayerIndex, u8 setupOpDisabled) {
+    u8 i;
+
+    for (i = 0; i < gActiveSeqs[seqPlayerIndex].setupCmdNum; i++) {
+        u8 setupOp = (gActiveSeqs[seqPlayerIndex].setupCmd[i] & 0xF00000) >> 20;
+
+        if (setupOp == setupOpDisabled) {
+            gActiveSeqs[seqPlayerIndex].setupCmd[i] = 0xFF000000;
+        }
+    }
+}
+
+void AudioSeq_SetVolumeScale(u8 seqPlayerIndex, u8 scaleIndex, u8 targetVol, u8 volFadeTimer) {
+    f32 volScale;
+    u8 i;
+
+    gActiveSeqs[seqPlayerIndex].volScales[scaleIndex] = targetVol & 0x7F;
+
+    if (volFadeTimer != 0) {
+        gActiveSeqs[seqPlayerIndex].fadeVolUpdate = true;
+        gActiveSeqs[seqPlayerIndex].volFadeTimer = volFadeTimer;
+    } else {
+        for (i = 0, volScale = 1.0f; i < VOL_SCALE_INDEX_MAX; i++) {
+            volScale *= gActiveSeqs[seqPlayerIndex].volScales[i] / 127.0f;
+        }
+
+        SEQCMD_SET_SEQPLAYER_VOLUME_NOW(seqPlayerIndex, volFadeTimer, volScale);
+    }
+}
+
+/**
+ * Update different commands and requests for active sequences
+ */
+void AudioSeq_UpdateActiveSequences(void) {
+    u32 tempoCmd;
+    u16 tempoPrev;
+    u16 seqId;
+    u16 channelMask;
+    u16 tempoTarget;
+    u8 setupOp;
+    u8 targetSeqPlayerIndex;
+    u8 setupVal2;
+    u8 setupVal1;
+    u8 tempoOp;
+    s32 pad[2];
+    u32 retMsg;
+    f32 volume;
+    u8 tempoTimer;
+    u8 seqPlayerIndex;
+    u8 j;
+    u8 channelIndex;
+
+    for (seqPlayerIndex = 0; seqPlayerIndex < SEQ_PLAYER_MAX; seqPlayerIndex++) {
+
+        // The seqPlayer has finished initializing and is currently playing the active sequences
+        if (gActiveSeqs[seqPlayerIndex].isSeqPlayerInit && gAudioCtx.seqPlayers[seqPlayerIndex].enabled) {
+            gActiveSeqs[seqPlayerIndex].isSeqPlayerInit = false;
+        }
+
+        // The seqPlayer is no longer playing the active sequences
+        if ((AudioSeq_GetActiveSeqId(seqPlayerIndex) != NA_BGM_DISABLED) &&
+            !gAudioCtx.seqPlayers[seqPlayerIndex].enabled && (!gActiveSeqs[seqPlayerIndex].isSeqPlayerInit)) {
+            gActiveSeqs[seqPlayerIndex].seqId = NA_BGM_DISABLED;
+        }
+
+        // Check if the requested sequences is waiting for fonts to load
+        if (gActiveSeqs[seqPlayerIndex].isWaitingForFonts) {
+            switch ((s32)AudioThread_GetExternalLoadQueueMsg(&retMsg)) {
+                case SEQ_PLAYER_BGM_MAIN + 1:
+                case SEQ_PLAYER_FANFARE + 1:
+                case SEQ_PLAYER_SFX + 1:
+                case SEQ_PLAYER_BGM_SUB + 1:
+                case SEQ_PLAYER_AMBIENCE + 1:
```

Comment:
Why the `+1`s?

Reply:
It's a long rabbit hole from the source to here, but the source comes from here:
https://github.com/engineer124/mm/blob/c4dab21fb75afc17eef8409b49f0d738e46df8fd/src/audio/sequence.c#L139